### PR TITLE
detect-caps-lock-is-on

### PR DESCRIPTION
**Pull request by Darin-Hamouda**
Adding JS Snippet: Detect Caps Lock Is On

### General Checks

- [ ] the branch is up to date with `main`/`master`
- [ ] the code works when pulled and run locally
- [ ] All CI checks pass
- [ ] all conflicts are resolved (if any)
- [ ] PR has a descriptive title
- [ ] PR has appropriate labels and milestones for easy identification
- [ ] PR it is assigned to the owner
- [ ] reviewers are assigned
- [ ] the PR contributes only one focused change
- [ ] It is in the appropriate column in the project board (if necessary)
- [ ] has short and clear description
- [ ] is linked to an issue (if it is related)
- [ ] feedback is addressed (if any and if it is appropriate feedback.)

### Markdown

- [ ] the markdown source is formatted
- [ ] spelling and grammar is correct in all text
- [ ] The markdown looks correct when you preview the file
- [ ] all links and images work
- [ ] any embedded code snippets are formatted

### CODE REVIEW

File: snippets/javascript/caps-lock-on.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+The following shows a warning message if you turn on the caps lock and type the
+password:
+
+    const password = document.querySelector('#password'); const message =
+    document.querySelector('.message');
+
+    password.addEventListener('keyup', function (e) { if
+    (e.getModifierState('CapsLock')) { message.textContent = 'Caps lock is on'; }
+    else { message.textContent = ''; } });
```
Comment (arnochauveau) [inline snippets/javascript/caps-lock-on.md:25]: Tip: you can use indentation to make your code more readable

```js
password.addEventListener('keyup', function (e) { 
    if (e.getModifierState('CapsLock')) {
        message.textContent = 'Caps lock is on'; 
    } 
    else { 
        message.textContent = ''; 
    }
});
```
As an extra tip: You could also take the function out of the addEventListener parameters like this to make it more readable: 

```js
password.addEventListener('keyup', setCapsLockMessage);

function setCapsLockMessage (event){
    if (event.getModifierState('CapsLock')) {
        message.textContent = 'Caps lock is on'; 
    } 
    else { 
        message.textContent = ''; 
    }
}
```

**Review by arnochauveau — CHANGES_REQUESTED** [review]
Very cool and useful snippet 🤩! 
Try to pay attention to indentation and formatting of your code as it is a big factor in readability!

File: README.md [hunk 2]
```diff
@@ -457,7 +457,6 @@ locally.
     </table> </li>
 </ul><br>
 
-
```
[the remark this answers is not in the document]
Comment (Darin-Hamouda) [reply README.md:460]: that file was changed by itself :( , I did not change any thing in it.
but I will revert the changes.

File: snippets/javascript/caps-lock-on.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Detect Caps Lock Is On
+
+To check if the caps lock is on, you use the `getModifierState()` method of the
+`KeyboardEvent` object:
+
+The `getModifierState()` method returns `true` if a modifier is active;
+otherwise, it returns `false`.
+
+The `event.getModifierState('CapsLock')` can be used to detect if the caps lock
+is on.
```
[the remark this answers is not in the document]
Comment (Darin-Hamouda) [reply snippets/javascript/caps-lock-on.md:10]: Thanks a lot

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (arnochauveau) [reply README.md:460]: A lot of people are having this problem, I'm going to take a look at it this evening too see if I can fix it.

File: snippets/javascript/caps-lock-on.md (hunk 1 above)
Comment (Darin-Hamouda) [reply snippets/javascript/caps-lock-on.md:25]: @arnochauveau 
I am trying to push the modifications, but I got this error when linting:
MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "````"]

I tried to place (```js ) before the code in the file, but it did not work.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Darin-Hamouda) [reply README.md:460]: @arnochauveau 
how to revert a single file from a commit?

[the remark this answers is not in the document]
Comment (arnochauveau) [reply README.md:460]: Hey Darin, i've added a fix for this so it won't happen again in the future!

You can revert the file by running `Git checkout origin/main -- README.md`

File: snippets/javascript/caps-lock-on.md (hunk 1 above)
Comment (arnochauveau) [reply snippets/javascript/caps-lock-on.md:25]: Did you add a closing ``` on the end of the code block? You can also try removing the first tab(whitespace before the lines) of the code block. Maybe that interferes.

Comment (Darin-Hamouda) [reply snippets/javascript/caps-lock-on.md]: @arnochauveau Finally worked, Thank you

**Review by arnochauveau — APPROVED** [review]
Hey @Darin-Hamouda, it was not 100% correct yet. But I fixed it for you as a christmas present 🙂. Take a look at my commits to see what you did wrong.

When using markdown, alignment, whitespacing and newlines are very important. 

```` 
```js
// your code here
```
````

or 
````
```html 
<!-- html code here -->
```
````

**Review by arnochauveau — APPROVED** [review]

**Pull request merged**